### PR TITLE
Ensure React component write effects do not conflict one another

### PR DESCRIPTION
**Pull request by trueadm**
Release notes: none

This PR adds some sanity around React components roots that when render, mutate the same of bindings as other React component roots. This is currently not supported and thus safely bails out and continues.

### CODE REVIEW

File: test/react/functional-components/simple-10.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <div>
+      <A />
+    </div>
+  );
+}
+
+App.getTrials = function(renderer, Root) {
+  // keep a reference to the other root that also
+  // writes to the same variable
+  let x = App2;
```
Comment (gaearon) [inline test/react/functional-components/simple-10.js]: What does this do?

Comment (trueadm) [reply test/react/functional-components/simple-10.js]: Holds a reference to the function or it gets DCE.

Comment (gaearon) [reply test/react/functional-components/simple-10.js]: Can we do something like `App.App2 = App2` in the body instead?

Comment (trueadm) [reply test/react/functional-components/simple-10.js]: Sure, I'm not concerned about how we reference it. I'll change.

**Pull request closed without merging**